File: leetcode/169.MajorityElement.go
package leetcode

func majorityElement(nums []int) int {
	result := 0
	count := 0

	for _, num := range nums {
		if count == 0 {
			result = num
		}
		if num == result {
			count++
		} else {
			count--
		}
	}

	return result
}

File: leetcode/169.MajorityElement.ts
function majorityElement(nums: number[]): number {
  let result = 0
  let count = 0

  for (const num of nums) {
    if (count === 0) {
      result = num
    }
    count = num === result ? count + 1 : count - 1
  }

  return result
}
console.log(majorityElement([3, 2, 3]))
console.log(majorityElement([2, 2, 1, 1, 1, 2, 2]))
